src/plugins/credential_management/public/components/edit_credential/edit_credential.tsx

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { withRouter, RouteComponentProps } from 'react-router-dom';
+import {
+  EuiHorizontalRule,
+  EuiGlobalToastList,
+  EuiGlobalToastListToast,
+  EuiForm,
+  EuiDescribedFormGroup,
+  EuiFormRow,
+  EuiFieldText,
+  EuiSelect,
+  EuiLink,
+  EuiButton,
+  EuiPageContent,
+  EuiFieldPassword,
+} from '@elastic/eui';
+import { DocLinksStart } from 'src/core/public';
+import { getCreateBreadcrumbs } from '../breadcrumbs';
+import { CredentialManagmentContextValue } from '../../types';
+// TODO: Add Header
+// import { Header } from './components/header';
+import { context as contextType } from '../../../../opensearch_dashboards_react/public';
+import { CredentialEditPageItem } from '../types';
+
+interface EditCredentialState {
+  credentialName: string;
+  credentialType: string;
+  userName: string;
+  password: string;
+  dual: boolean;
+  toasts: EuiGlobalToastListToast[];
+  docLinks: DocLinksStart;
+}
+
+export interface EditCredentialProps extends RouteComponentProps {
+  credential: CredentialEditPageItem;
+}
+
+export class EditCredentialComponent extends React.Component<
+  EditCredentialProps,
+  EditCredentialState
+> {
+  static contextType = contextType;
+  public readonly context!: CredentialManagmentContextValue;
+  constructor(props: EditCredentialProps, context: CredentialManagmentContextValue) {
+    super(props, context);
+
+    context.services.setBreadcrumbs(getCreateBreadcrumbs());
+
+    this.state = {
+      credentialName: props.credential.title,
+      credentialType: props.credential.credentialType,
+      userName: '',
+      password: '',
+      dual: true,
+      toasts: [],
+      docLinks: context.services.docLinks,
+    };
+  }
+
+  // TODO: Add conditional rendering to select credential types
+  renderContent() {
+    const options = [
+      { value: 'username_password_credential', text: 'Username and Password Credential' },
+      { value: 'no_auth', text: 'No Auth' },
+    ];
+
+    return (
```

Comment:
Will probably require i18n support, might be easier to do it along the way

Reply:
> i18n support

Yes, more i18n support is required. Because this is the initial PR and we are iterating quickly. As a quick follow-up, I'd like to create an issue: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2047